### ISSUE TITLE
[java] False positive for `UseConcurrentHashMap` on `ConcurrentSkipListMap`

### ISSUE THREAD

**Issue by vbrandl** [pmd/pmd#2786]
**Affects PMD Version:** 6.27.0

**Rule:**

[`UseConcurrentHashMap`](https://pmd.github.io/pmd-6.27.0/pmd_rules_java_multithreading.html#useconcurrenthashmap)

**Description:**

When using a `ConcurrentSkipListMap` (to replace a `TreeMap`), PMD still gives a `UseConcurrentHashMap` lint.

**Code Sample demonstrating the issue:**

```java
import java.util.Map;
import java.util.concurrent.ConcurrentSkipListMap;

public class App {
    public final Map<String, String> m = new ConcurrentSkipListMap<>();
    
    public App() {
        System.err.println("hello world");
    }
}

```

**Expected outcome:**

This should not produce a warning, since `ConcurrentSkipListMap` is a `ConcurrentMap`

**Running PMD through:** *Gradle*
